Bad read splice_insert: 16 expected: 20

When I parse this SCTE ("/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg==") or any others from our stream, I get a parsed response from the library, but also get this error in the console:
```
Bad read splice_insert actual: 16 expected: 20
spliceInsert	@	scte35.js:111
parseSCTE35Data	@	scte35.js:166
push../node_modules/scte35/build/lib/scte35.js.SCTE35.parseFromB64	@	scte35.js:194
```
Running the same cue in the CLI on master, I get:
```
$ npm run parse "/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg=="

> scte35@0.2.0 parse C:\_libraries\scte35
> node ./scripts/cli.js "C:/Program Files/Git/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg=="


C:\_libraries\scte35\lib\scte35.ts:284
        const splice = new DataView(bytes.buffer, offset, sis.spliceCommandLength);
                       ^
RangeError: Invalid DataView length undefined
    at new DataView (<anonymous>)
    at parseSCTE35Data (C:\_libraries\scte35\lib\scte35.ts:284:24)
    at Object.exports.SCTE35.parseFromB64 (C:\_libraries\scte35\lib\scte35.ts:322:12)
    at Object.<anonymous> (C:\_libraries\scte35\scripts\cli.js:22:42)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! scte35@0.2.0 parse: `node ./scripts/cli.js "C:/Program Files/Git/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg=="`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the scte35@0.2.0 parse script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Nils\AppData\Roaming\npm-cache\_logs\2019-05-23T16_35_58_717Z-debug.log
```
I also tried on the splice-descriptors branch and got this error:
```
$ npm run parse "/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg=="

> scte35@0.2.0 parse C:\_libraries\scte35
> node ./scripts/cli.js "C:/Program Files/Git/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4                                                                                                                                  AUmXAAAAAAAAAdIQsGg=="


C:\_libraries\scte35\lib\scte35.ts:266
        throw new Error(`Binary read error sectionLength: ${sis.sectionLength} +                                                                                                                                   3 !== data.length: ${bytes.byteLength}`);
              ^
Error: Binary read error sectionLength: 1003 + 3 !== data.length: 54
    at parseSCTE35Data (C:\_libraries\scte35\lib\scte35.ts:266:15)
    at Object.exports.SCTE35.parseFromB64 (C:\_libraries\scte35\lib\scte35.ts:34                                                                                                                                  4:12)
    at Object.<anonymous> (C:\_libraries\scte35\scripts\cli.js:22:35)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! scte35@0.2.0 parse: `node ./scripts/cli.js "C:/Program Files/Git/DAlAAA                                                                                                                                  AAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg=="`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the scte35@0.2.0 parse script.
npm ERR! This is probably not a problem with npm. There is likely additional log                                                                                                                                  ging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Nils\AppData\Roaming\npm-cache\_logs\2019-05-23T16_35_15_6                                                                                                                                  29Z-debug.log
```

--- a/lib/scte35.ts
+++ b/lib/scte35.ts
@@ -75,7 +75,7 @@ const spliceEvent = (event: SpliceEvent, view: DataView, tag: EventTag): number 
 
     if (event.programSpliceFlag) {
         if (tag === SpliceCommandType.SPLICE_INSERT && !(event as ISpliceInsertEvent).spliceImmediateFlag) {
-            const spliceTime = timeSignal(new DataView(view.buffer, offset, 5));
+            const spliceTime = timeSignal(new DataView(view.buffer, view.byteOffset + offset, 5));
             (event as ISpliceInsertEvent).spliceTime = spliceTime;
             offset++;
             if (spliceTime.specified) {

--- a/lib/scte35.ts
+++ b/lib/scte35.ts
@@ -144,7 +144,7 @@ const spliceSchedule = (view: DataView): ISpliceSchedule => {
 
     while (schedule.spliceEvents.length !== schedule.spliceCount) {
         const event = {} as ISpliceScheduleEvent;
-        offset += spliceEvent(event, new DataView(view.buffer, offset), SpliceCommandType.SPLICE_SCHEDULE);
+        offset += spliceEvent(event, new DataView(view.buffer, view.byteOffset + offset), SpliceCommandType.SPLICE_SCHEDULE);
         schedule.spliceEvents.push(event);
     }
     if (offset !== view.byteLength) {
@@ -267,7 +267,7 @@ const parseSCTE35Data = (bytes: Uint8Array): ISpliceInfoSection => {
     }
     // NOTE(estobb200): Can't shift JavaScript numbers above 32 bits
     sis.ptsAdjustment = (byte & 0x01) ? THIRTY_TWO_BIT_SHIFT : 0;
-    sis.ptsAdjustment += view.getUint32(offset)
+    sis.ptsAdjustment += view.getUint32(offset);
     offset += 4;
 
     sis.cwIndex = view.getUint8(offset++);

--- a/test/scte35.test.ts
+++ b/test/scte35.test.ts
@@ -64,6 +64,7 @@ describe("SCTE35", () => {
         const hex = "/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg==";
         const spliceInfo = scte35.SCTE35.parseFromB64(hex);
         chai.expect((spliceInfo.spliceCommand as any).spliceTime.specified).to.eql(true);
+        chai.expect((spliceInfo.spliceCommand as any).breakDuration.duration).to.eql(5400000);
     });
 
 });

--- a/test/scte35.test.ts
+++ b/test/scte35.test.ts
@@ -60,4 +60,10 @@ describe("SCTE35", () => {
         chai.expect(spliceInfo.ptsAdjustment).to.eq(4629503913);
     });
 
+    it("should parse splice_insert correctly", () => {
+        const hex = "/DAlAAAAAAAAAP/wFAUAAqbVf+/+AAAAAH4AUmXAAAAAAAAAdIQsGg==";
+        const spliceInfo = scte35.SCTE35.parseFromB64(hex);
+        chai.expect((spliceInfo.spliceCommand as any).spliceTime.specified).to.eql(true);
+    });
+
 });

